sql/2024/cms/core_web_vitals_yoy.sql

```diff
@@ -0,0 +1,173 @@
+# cms passing core web vitals
+# core_web_vitals_yoy.sql
+CREATE TEMP FUNCTION IS_GOOD (good FLOAT64, needs_improvement FLOAT64, poor FLOAT64) RETURNS BOOL AS (
+  good / (good + needs_improvement + poor) >= 0.75
+);
+
+CREATE TEMP FUNCTION IS_NON_ZERO (good FLOAT64, needs_improvement FLOAT64, poor FLOAT64) RETURNS BOOL AS (
+  good + needs_improvement + poor > 0
+);
+
+SELECT
+  2024 AS year,
+  client,
+  cms,
+  COUNT(DISTINCT origin) AS origins,
+  # Origins with good LCP divided by origins with any LCP.
+  SAFE_DIVIDE(
+    COUNT(DISTINCT IF(IS_GOOD(fast_lcp, avg_lcp, slow_lcp), origin, NULL)),
+    COUNT(DISTINCT IF(IS_NON_ZERO(fast_lcp, avg_lcp, slow_lcp), origin, NULL))) AS pct_good_lcp,
+
+  # Origins with good INP divided by origins with any INP.
+  SAFE_DIVIDE(
+    COUNT(DISTINCT IF(IS_GOOD(fast_inp, avg_inp, slow_inp), origin, NULL)),
+    COUNT(DISTINCT IF(IS_NON_ZERO(fast_inp, avg_inp, slow_inp), origin, NULL))) AS pct_good_inp,
+  # Origins with good CLS divided by origins with any CLS.
+  SAFE_DIVIDE(
+    COUNT(DISTINCT IF(IS_GOOD(small_cls, medium_cls, large_cls), origin, NULL)),
+    COUNT(DISTINCT IF(IS_NON_ZERO(small_cls, medium_cls, large_cls), origin, NULL))) AS pct_good_cls,
+
+  # Origins with good LCP, INP (optional), and CLS divided by origins with any LCP and CLS.
+  SAFE_DIVIDE(
+    COUNT(DISTINCT IF(
+      IS_GOOD(fast_lcp, avg_lcp, slow_lcp) AND
+      IS_GOOD(fast_inp, avg_inp, slow_inp) IS NOT FALSE AND
+      IS_GOOD(small_cls, medium_cls, large_cls), origin, NULL)),
+    COUNT(DISTINCT IF(
+      IS_NON_ZERO(fast_lcp, avg_lcp, slow_lcp) AND
+      IS_NON_ZERO(small_cls, medium_cls, large_cls), origin, NULL))) AS pct_good_cwv
+FROM
+  `chrome-ux-report.materialized.device_summary`
+JOIN (
+  SELECT
+    client,
+    page url,
```

Comment:
I don't know how to address this @tunetheweb any suggestions?  
 

```
== [/github/workspace/sql/2024/cms/core_web_vitals_yoy.sql] FAIL
223L: 44 | P: 10 | L012 | Implicit/explicit aliasing of columns.
224
```

Reply:
Should be `page AS url` rather than `page url`.

The linter prefers you are explicit about this being an alias by using `AS`. Otherwise it could be just a typo due missing a comma and meant to be two columns: `SELECT page, url`.

Reply:
Right..  Thanks..